src/lib.rs

```diff
@@ -9,6 +9,94 @@
 // disable warnings in that scenario.
 #![cfg_attr(not(rayon_unstable), allow(warnings))]
 
+//! Data-parallelism library that is easy to convert sequential computations into parallel.
+//!
+//! Rayon is lightweight and convenient for application to existing code.  It guarantees
+//! data-race free executions, and takes advantage of parallelism when sensible, based
+//! on work-load at runtime.  There are two categories of rayon workloads: parallel
+//! iterators and multi-branched recursion (`join` method).
+//!
+//! # Parallel Iterators
+//!
+//! Parallel iterators are formed using [`par_iter`] and `par_iter_mut` functions
+//! followed by a computation chain.  Computations on the iterator can take the
+//! shape of `map` or `for_each` as an example.  This solves [embarrassingly]
+//! parallel tasks that are completely independent of one another.
+//!
+//! [`par_iter`]: iter/trait.ParallelIterator.html
+//! [embarrassingly]: https://en.wikipedia.org/wiki/Embarrassingly_parallel
+//!
+//! # Examples
+//!
+//! ```
+//! extern crate rayon;
+//! use rayon::prelude::*;
+//! # fn main() {
+//! let mut chars: Vec<char> = "A man, a plan, a canal - Panama!".chars().collect();
+//! let encrypted: String = chars.par_iter().map(|c| {
+//!        match *c {
+//!            'A' ... 'M' | 'a' ... 'm' => ((*c as u8) + 13) as char,
+//!            'N' ... 'Z' | 'n' ... 'z' => ((*c as u8) - 13) as char,
+//!            _ => *c
+//!        }
+//!    }).collect();
+//! # }
```

Comment:
It would be good to `assert_eq!` the result so readers know what to expect, and so it also serves as a doctest.